Dikubot/Webapp/Shared/Facebook/FacebookAddPages.razor

```diff
@@ -0,0 +1,144 @@
+
+    //return ;
+        }
+        return "";
+    }
+    
+
+    
+    
+    private void LoginFacebookAction()
+    {
+        string link =
+            "https://www.facebook.com/v16.0/dialog/oauth?" +
+            "client_id=" + Environment.GetEnvironmentVariable("FACEBOOK_CLIENT_ID") + //510045277999983
+            "&redirect_uri="+_navigationManager.BaseUri +"/FacebookAddPages" +
+            "&auth_type=rerequest" +
+            "&config_id=1372246313628981";
+            // Environment.GetEnvironmentVariable("DISCORD_CLIENT_ID")
+        
+        _navigationManager.NavigateTo(link);
+    }
+
+    protected override async void OnInitialized()
+    {
+        if (_navigationManager.Uri.Contains("code"))
+        {
+            string header = _navigationManager.Uri.Remove(0, _navigationManager.Uri.IndexOf('?') + 1);
+            List<string> Request = header.Split('&', '=').ToList();
+
+    //error=access_denied&error_description=The+resource+owner+or+authorization+server+denied+the+request
+
+            
+            var app_id = Environment.GetEnvironmentVariable("FACEBOOK_CLIENT_ID");//"510045277999983";
+            var redirect_uri = _navigationManager.BaseUri+"/FacebookAddPages";
+            var app_secret= Environment.GetEnvironmentVariable("FACEBOOK_APP_SECRET");//"a273d3f3518a451d1378e678001acce3";
+            var code_parameter = Request[Request.FindIndex(x => x.Equals("code")) + 1];;
+            var graph_api_version= "v16.0";
+            
+            var link = $"https://graph.facebook.com/{graph_api_version}/oauth/access_token?client_id={app_id}&redirect_uri={redirect_uri}&client_secret={app_secret}&code={code_parameter}=_";
+            
+            FBAccessToken fbAccessToken = HttpClient.GetFromJsonAsync<FBAccessToken>(link).Result;
+
+            AuthCode = fbAccessToken.access_token;
+            var link2 = $"https://graph.facebook.com/v16.0/me/accounts?access_token={AuthCode}";
+            
+            FBresponsList = HttpClient.GetFromJsonAsync<FBrespons>(link2).Result.data;
+
+            var link3 = $"https://graph.facebook.com/v16.0/oauth/access_token?grant_type=fb_exchange_token&" +
+                        $"client_id={app_id}&" +
+                        $"client_secret={app_secret}&" +
+                        $"fb_exchange_token={fbAccessToken.access_token}";
+           var respons3 = HttpClient.GetFromJsonAsync<FBAccessToken>(link3).Result;
+
+            var link4 = $"https://graph.facebook.com/me?fields=id&access_token={respons3.access_token}";
+            var app_scoped_user_id = HttpClient.GetFromJsonAsync<FBidRespons>(link4).Result.id;
+            
+            var link5 = $"https://graph.facebook.com/{graph_api_version}/{app_scoped_user_id}/" +
+                        $"accounts?access_token={respons3.access_token}";
+
+            var test5 = HttpClient.GetFromJsonAsync<FBrespons>(link5).Result.data;
+            
```

Comment:
The variable names don't follow the C# naming convention, also they are not very helpful. Try to be descriptive in the names. What is link5, which uses response3? It is hard to read.